### PR TITLE
Greys make Grey Chairs/Beds

### PR DESCRIPTION
**Pull request by ghost**
## What this does
Greys will now make the Mothership crates/beds/chairs. Direct switch for crates and closets. They make the non-premium bed and chairs. Just the beige chair though, they still make the others the human way.
Copypasted the code and did my best. I copypasted from Dreammaker the changes and I think I got everything but I may be wrong. 

## Why it's good
A little bit of flavor for non-human races and a way to see the Mothership stuff on station more often.

## Changelog

:cl:
 * rscadd: Greys now make their own crates/beds/chairs/closets after training from the Mothership.
 * rscdel: Greys no longer make the normal versions of the above.

### CODE REVIEW

**Comment by Kurfursten** [comment]
Did you test your code? Because looking at this code, it looks like this will prevent greys from making most objects through crafting recipes. Since the grey output is null by default, any recipe that has no grey output should create null! For example, if they want to craft a barrel or a railing. Also, this is extremely snowflaky. Tie it to the species datum in a modular fashion.

Besides the coding issues, I'm not sure I like this conceptually. Maybe it would be best to put all the grey stuff (walls, floors, crates, chairs) in a special gray metal.

**Comment by ghost** [comment]
You're right, I only tested the making of the closet/crate stuff when I went to add the beds/chairs. This does prevent greys from making the other stuff so I'll be going back and trying to fix it

**Comment by ErrorsWindows** [comment]
:pensive:

**Comment by ghost** [comment]
I think it works now? Greys can make their unique stuff now and making other non-grey stuff is fixed now.  Still has the problem of being snowflakey but it works and I'm a novice. I'd be more than happy to tie it to species datums and make it more modular if you can point me toward something where I could get an idea of just how to do that. As for the concept itself, I'm a fan of the idea that alien races make their own unique items that aren't too far off the standard set by humans and I got a couple positive responses in the thread to the idea of the crates.

**Comment by SonixApache** [comment]
spitballing: add teflon-coated metal and a way to get it, move grayyy stuff to teflon-coated metal
that said i really do like the idea of being able to produce species-specific stuff, gives species more flair

File: code/game/objects/items/stacks/stack_recipes.dm [hunk 1]
```diff
@@ -95,7 +97,10 @@
 		SS.update_materials()
 	else
 		for(var/i = 1 to (max_res_amount>1 ? res_amount*multiplier : 1))
-			O = new result_type(usr.loc)
+			if(isgrey(usr) && grey_result <> null)
```
Comment (Kurfursten) [inline code/game/objects/items/stacks/stack_recipes.dm:100]: != is the not equal operator

**Comment by Kurfursten** [comment]
> I'd be more than happy to tie it to species datums and make it more modular if you can point me toward something where I could get an idea of just how to do that. 

One way to do it would be to create a reference list inside the species datum with a proc to call. For example...

```
/datum/species/proc/replace_recipe_result(var/output_type)
  return output_type

/datum/species/gray/replace_recipe_result(var/output_type)
  switch(output_type) //search for exact type match to replace
    if(/obj/structure/bed/chair/comfy/beige)
       return /obj/structure/bed/chair/comfy/ayy1
    if(/obj/structure/bed
       return /obj/structure/bed/ayy1)
  //No exact type match, use original
  return output_type
```

You can then call this to handle the type substitution; this would allow you to easily swap in new recipes for any race and store them in the datum. I still prefer the teflon idea though, and I think it's weird that grays can make other comfy chairs just not beige ones.

Another way would be to not do type substitution at all but render a different list based on the species that makes it. Add a new var that is a list of permitted species. Then when the recipe list is displayed, display only the ones that your race can make. This would let you blacklist grays from making, for example, ALL comfy chairs while adding just one new chair for grays.

**Comment by Securitymain** [comment]
I didnt look at the code but the idea itself is great and if implemented well and modular(so implementation for other races follows the same pattern) this will be great

**Comment by Eneocho** [comment]
Not liking that I be forced to play ayylmaos to make the cool chairs/beds.

**Comment by ghost** [comment]
@Kurfursten Thank you for that example
I think from here I'll try a mix of both. Type substitution for the crate/closet/bed/ in the species datum and then permitted species var so I can add the chair/premiumchair/premiumbed as their own recipes. The beige chair was an unfortunate hit since the bare metal chair didn't seem like a good switch for the Ayy chair sprite and it was the next best thing.  I just have to find out how to go about hiding the recipes since I'm not sure if I do that in the recipes here or the stack.dm which seems a bit more harder to read.

**Comment by Kurfursten** [comment]
It's stack.dm

Around ~L81 it already does a continue to exit if the map doesn't allow z-level buildings like ladders or stairs. You can add a new if statement to check race restrictions and then simply not render the line at all if it's not in the racial whitelist. I would do it something like

```
if(R.species_whitelist.len) //This is a whitelisted object, default behavior is no whitelist
  //Only humans can have species, so kick out anything that isn't at least human
  if(!ishuman(user))
    continue
  //Casting user as a human to get its species
  var/mob/living/carbon/human/H = user
  if(!(H.species in R.species_whitelist))
    continue
  //Exit if not in the whitelist, otherwise continue
```

**Comment by ManopolisJones** [comment]
I don't agree with this, for the practical reason that you can't rig electronics to the grayy crates (you can stick wire in them, but not the actual airlock electronics). For fluff reasons, i'd probably say these kind of lockers and crates are just slightly more sophisticated then being able to morph them with metal, I know that may sound retarded considering we can make a bunch of shit with metal like it's no thing because we're in a video game, but my point is, i'd rather have some kind of tool to make these mothership constructions then have it be race limited.

**Comment by ghost** [comment]
After trying to get the recipe list to only render for Greys to no success, I'll be closing this and may give it a shot at a later date. I also thought the closets/crates could accept circuits to secure them but it seems that's not the case. The closet has a secure variant so that's easy enough to make it accept the circuit but the crate doesn't and there doesn't seem to be a sprite lying around for one to be added

**Comment by Kurfursten** [comment]
If you wanted to do the implementation with teflon metal that would be extremely easy - make a separate recipe list for the teflon and you're all set. All you'd need is a sprite and a way to get teflon. My recommendation?

> Mix fluorine and carbon in a beaker (possibly heat it up?) to produce the new material, like how plastic is made with acid and plasticide

This feels very grey thematic to me. Making your couches in a beaker.

**Pull request closed without merging**